lib/daru/vector.rb

```diff
@@ -762,25 +762,25 @@ def replace_nils! replacement
 
     # Lags the series by k periods.
     #
-    # The convention is to set the oldest observations (the first ones
-    # in the series) to nil so that the size of the lagged series is the
-    # same as the original.
+    # Lags the series by k periods where k can be a positive or negative
+    # integer.
+    # Returns a new series with nil values inserted into the series
+    # keeping the size of the new series the same as the original.
     #
     # Usage:
     #
-    #   ts = Daru::Vector.new((1..10).map { rand })
-    #           # => [0.69, 0.23, 0.44, 0.71, ...]
+    #   ts = Daru::Vector.new(1..5)
+    #               # => [1, 2, 3, 4, 5]
+    #
+    #   ts.lag      # => [nil, 1, 2, 3, 4]
+    #   ts.lag(1)   # => [nil, 1, 2, 3, 4]
+    #   ts.lag(2)   # => [nil, nil, 1, 2, 3]
+    #   ts.lag(-1)  # => [2, 3, 4, 5, nil]
     #
-    #   ts.lag   # => [nil, 0.69, 0.23, 0.44, ...]
-    #   ts.lag(2) # => [nil, nil, 0.69, 0.23, ...]
     def lag k=1
       return dup if k.zero?
-
-      dat = @data.to_a.dup
-      (dat.size - 1).downto(k) { |i| dat[i] = dat[i - k] }
-      (0...k).each { |i| dat[i] = nil }
-
-      Daru::Vector.new(dat, index: @index, name: @name)
+      return Daru::Vector.new([nil]*@data.size, index: @index, name: @name) if k.abs >= @data.size
+      _lag k
     end
```

Comment:
WDYT about this implementation? Rubocop is happy, and instead of strangely named method "just to fool rubocop" we got internal useful method (create vector with a new data, and same index/name as a current):
```ruby
    def lag k=1
      case k
      when 0 then dup
      when 1...size
        copy([nil] * k + data.to_a)
      when -size..-1
        copy(data.to_a[k.abs...size])
      else
        copy([])
      end
    end

    private 
    def copy(values)
      # Make sure values is right-justified to the size of the vector
      values.concat([nil] * (size - values.size)) if values.size < size
      Daru::Vector.new(values[0...size], index: @index, name: @name)
    end
```